### PR TITLE
[BM-511] Firebase analytics

### PR DESCRIPTION
**Pull request by gabrysgab**
### Task

 - [JIRA](https://netguru.atlassian.net/browse/BM-511)
 
### Description

Added Firebase performance and firebase analytics events.

### CODE REVIEW

File: app/src/main/kotlin/co/netguru/baby/monitor/client/common/base/BaseFragment.kt [hunk 1]
```diff
@@ -2,15 +2,27 @@ package co.netguru.baby.monitor.client.common.base
...
 
-abstract class BaseFragment : Fragment() {
+abstract class BaseFragment : DaggerFragment(), AnalyticScreen {
     abstract val layoutResource: Int
+    override val screenName: String? = null
+
+    @Inject
+    lateinit var analyticsManager: AnalyticsManager
 
     override fun onCreateView(
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/common/base/BaseFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/common/base/BaseFragment.kt#L18"></span>Parameter should be on a separate line (unless all parameters can fit a single line)</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
+package co.netguru.baby.monitor.client.feature.analytics
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt#L1"></span>Needless blank line(s)</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+import android.app.Activity
+import android.os.Bundle
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt#L4"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    private val firebaseAnalytics: FirebaseAnalytics
+) {
+    fun setCurrentScreen(activity: Activity, screenName: String) {
+        firebaseAnalytics.setCurrentScreen(activity, screenName, null)
+        Timber.d("screen $screenName")
+    }
+
+    fun logEvent(eventName: String) {
+        firebaseAnalytics.logEvent(eventName, null)
+        Timber.d("event $eventName")
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt#L19"></span>Multiple occurrences of the same string literal within a single file detected.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    fun logEventWithParam(eventName: String, pair: Pair<String, Any>) {
+        firebaseAnalytics.logEvent(eventName, bundleOf(pair))
+        Timber.d("event $eventName")
+    }
+
+    companion object {
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt#L27"></span>Missing space after //</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        const val NOTIFICATION_SNOOZE_EVENT = "notification_snooze"
+        const val NOTIFICATION_OPEN_CAMERA_EVENT = "notification_open_camera"
+
+        const val RESET_APP_EVENT = "reset_app"
+
+        const val NIGHT_MODE_EVENT = "night_mode"
+        const val ENABLED_PARAM = "is_enabled"
+
+        const val VIDEO_STREAM_CONNECTED = "video_stream_connected"
+        const val VIDEO_STREAM_ERROR = "video_stream_error"
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt#L41"></span>Missing space after //</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+        //Screen names
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt#L43"></span>Missing space after //</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        //General
+        const val SPLASH = "Splash"
+        const val ONBOARDING = "Onboarding"
+        const val INFO_ABOUT_DEVICES = "InfoAboutDevices"
+        const val SPECIFY_DEVICE = "SpecifyDevice"
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt#L48"></span>Missing space after //</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+        //Child Device
+        const val VOICE_RECORDINGS_SETTING = "VoiceRecordingsSetting"
+        const val CONNECT_WIFI = "ConnectWifi"
+        const val PERMISSION_CAMERA = "PermissionCamera"
+        const val PERMISSION_MICROPHONE = "PermissionMicrophone"
+        const val PERMISSION_DENIED = "PermissionDenied"
+        const val SETUP_INFORMATION = "SetupInformation"
+        const val CHILD_MONITOR = "ChildMonitor"
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt#L56"></span>Missing space after //</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+        //Parent Device
+        const val PARENT_DEVICE_INFO = "ParentDeviceInfo"
+        const val SERVICE_DISCOVERY = "ServiceDiscovery"
+        const val PAIRING = "Pairing"
+        const val CONNECTION_FAILED = "ConnectionFailed"
+        const val ALL_DONE = "AllDone"
+        const val CLIENT_LIVE_CAMERA = "ClientLiveCamera"
+        const val CLIENT_DASHBOARD = "ClientDashboard"
+        const val CLIENT_ACTIVITY_LOG = "ClientActivityLog"
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt#L65"></span>Unexpected blank line(s) before “}”</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/babycrynotification/OpenCameraUseCase.kt [hunk 11]
```diff
@@ -0,0 +1,26 @@
+package co.netguru.baby.monitor.client.feature.babycrynotification
+
+import android.os.Bundle
+import androidx.core.os.bundleOf
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/babycrynotification/OpenCameraUseCase.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/babycrynotification/OpenCameraUseCase.kt#L4"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/pairing/ServiceDiscoveryFragment.kt [hunk 12]
```diff
@@ -27,8 +28,9 @@ import kotlinx.android.synthetic.main.fragment_connecting_devices.*
 import java.util.concurrent.TimeUnit
 import javax.inject.Inject
 
-class ServiceDiscoveryFragment : BaseDaggerFragment() {
+class ServiceDiscoveryFragment : BaseFragment() {
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/pairing/ServiceDiscoveryFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/pairing/ServiceDiscoveryFragment.kt#L31"></span>Class ‘ServiceDiscoveryFragment’ with ‘15’ functions detected. Defined threshold inside classes is set to ‘15’</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/firebasenotification/FirebaseNotificationSender.kt [hunk 13]
```diff
@@ -1,8 +1,12 @@
 package co.netguru.baby.monitor.client.feature.firebasenotification
 
+import androidx.core.os.bundleOf
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/firebasenotification/FirebaseNotificationSender.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/firebasenotification/FirebaseNotificationSender.kt#L3"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/onboarding/InfoAboutDevicesFragment.kt [hunk 14]
```diff
@@ -6,10 +6,12 @@ import android.view.View
 import androidx.navigation.fragment.findNavController
 import co.netguru.baby.monitor.client.R
 import co.netguru.baby.monitor.client.common.base.BaseFragment
+import co.netguru.baby.monitor.client.feature.analytics.AnalyticsManager.Companion.INFO_ABOUT_DEVICES
 import kotlinx.android.synthetic.main.fragment_info_about_devices.*
 
 class InfoAboutDevicesFragment: BaseFragment() {
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/onboarding/InfoAboutDevicesFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/onboarding/InfoAboutDevicesFragment.kt#L12"></span>Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/settings/ServerSettingsFragment.kt [hunk 15]
```diff
@@ -10,13 +10,14 @@ import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.ViewModelProviders
 import co.netguru.baby.monitor.client.BuildConfig
 import co.netguru.baby.monitor.client.R
-import co.netguru.baby.monitor.client.common.base.BaseDaggerFragment
+import co.netguru.baby.monitor.client.common.base.BaseFragment
+import co.netguru.baby.monitor.client.feature.onboarding.OnboardingActivity
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/settings/ServerSettingsFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/settings/ServerSettingsFragment.kt#L14"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: app/src/main/res/layout/fragment_client_activity_log.xml [hunk 16]
```diff
@@ -1,28 +1,34 @@
...
-    android:layout_height="match_parent"
-    android:orientation="vertical"
-    android:background="@color/background">
+<LinearLayout
+        xmlns:android="http://schemas.android.com/apk/res/android"
+        xmlns:app="http://schemas.android.com/apk/res-auto"
+        xmlns:tools="http://schemas.android.com/tools"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:background="@color/background"
```
Comment (netguru-hound) [inline app/src/main/res/layout/fragment_client_activity_log.xml]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/res/layout/fragment_client_activity_log.xml#L8"></span>Possible overdraw: Root element paints background <code>@color/background</code> with a theme that also paints a background (inferred theme is <code>@style/AppTheme</code>)</td>
    </tr>
  </tbody>
</table>

**Comment by netguru-hound** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          5 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/application/FragmentBindingsModule.kt#L25">app/src/main/kotlin/co/netguru/baby/monitor/client/application/FragmentBindingsModule.kt#L25</a> - Class ‘FragmentBindingsModule’ with ‘21’ functions detected. Defined threshold inside classes is set to ‘15’</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/client/home/livecamera/ClientLiveCameraFragmentViewModel.kt#L42">app/src/main/kotlin/co/netguru/baby/monitor/client/feature/client/home/livecamera/ClientLiveCameraFragmentViewModel.kt#L42</a> - Unexpected indentation (expected 4, actual 8)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/ChildMonitorFragment.kt#L178">app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/ChildMonitorFragment.kt#L178</a> - The function serverViewModelObservers appears to be too complex.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/ChildMonitorFragment.kt#L237">app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/ChildMonitorFragment.kt#L237</a> - Unexpected blank line(s) before “}”</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/netguru/baby-monitor-client-android/blob/a540cb343de5354d91c6cd7978445c3390e2e9c1/app/src/main/res/layout/fragment_parent_device_info.xml#L198">app/src/main/res/layout/fragment_parent_device_info.xml#L198</a> - Missing <code>contentDescription</code> attribute on image</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">:link: <a href="https://netguru.atlassian.net/browse/BM-511">BM-511</a></td>
    </tr>
  </tbody>
</table>

### JaCoCO Code Coverage 12.62% :white_check_mark:
| Class | Covered | Meta | Status |
|:---|:---:|:---:|:---:|
| `co/netguru/baby/monitor/client/feature/welcome/SpecifyDeviceFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/client/home/log/ClientActivityLogFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/babycrynotification/OpenCameraUseCase` | 100% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/babycrynotification/CryingActionIntentService` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/babycrynotification/SnoozeNotificationUseCase` | 100% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/client/home/dashboard/ClientDashboardFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/common/base/BaseFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/settings/ServerSettingsFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/settings/ClientSettingsFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/settings/ResetAppUseCase` | 100% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/server/ChildMonitorViewModel` | 50% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/server/ChildMonitorFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/application/ApplicationModule` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/application/FragmentBindingsModule` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/communication/pairing/ServiceDiscoveryFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/communication/pairing/PairingFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/communication/pairing/ConnectingDevicesFailedFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/client/home/livecamera/ClientLiveCameraFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/client/home/livecamera/ClientLiveCameraFragmentViewModel` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/onboarding/OnboardingActivity` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/onboarding/FeaturePresentationFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/onboarding/InfoAboutDevicesFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/onboarding/VoiceRecordingsSettingsFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/splash/SplashFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/client/configuration/ParentDeviceInfoFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/client/configuration/AllDoneFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/firebasenotification/FirebaseNotificationSender` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/onboarding/baby/PermissionDenied` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/onboarding/baby/PermissionCameraFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/onboarding/baby/SetupInformationFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/onboarding/baby/PermissionMicrophoneFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/onboarding/baby/ConnectWifiFragment` | 0% | 0% | :white_check_mark: |


#### APK file size

| size |
| --- |
| 30.4MB |



#### Number of method references

| Dex file | count |
| --- | --- |
| classes.dex | 46053 |



<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+    private val firebaseAnalytics: FirebaseAnalytics
+) {
+    fun setCurrentScreen(activity: Activity, screenName: String) {
+        firebaseAnalytics.setCurrentScreen(activity, screenName, null)
+        Timber.d("$SCREEN $screenName")
+    }
+
+    fun logEvent(eventName: String) {
+        firebaseAnalytics.logEvent(eventName, null)
+        Timber.d("$EVENT $eventName")
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/3bf76ac46cfa16ed8786cf59815392e8654387bd/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt#L18"></span>Multiple occurrences of the same string literal within a single file detected.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt [hunk 18]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    fun logEventWithParam(eventName: String, pair: Pair<String, Any>) {
+        firebaseAnalytics.logEvent(eventName, bundleOf(pair))
+        Timber.d("$EVENT $eventName")
+    }
```
Comment (Aruziell) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt]: You could refactor these methods into a single one, with an optional parameter defaulting to null.

Comment (Aruziell) [reply app/src/main/kotlin/co/netguru/baby/monitor/client/feature/analytics/AnalyticsManager.kt]: I would think about typing `eventName` argument with an `enum` if you don't require arbitrary strings to be permitted?

File: app/src/main/kotlin/co/netguru/baby/monitor/client/common/base/AnalyticScreen.kt [hunk 19]
```diff
@@ -0,0 +1,5 @@
+package co.netguru.baby.monitor.client.common.base
+
+interface AnalyticScreen {
+    val screenName: String?
```
Comment (Aruziell) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/common/base/AnalyticScreen.kt]: How about using `enum` in place of screen name strings?

File: app/src/main/kotlin/co/netguru/baby/monitor/client/common/base/BaseFragment.kt [hunk 20]
```diff
@@ -2,15 +2,28 @@ package co.netguru.baby.monitor.client.common.base
...
 
-abstract class BaseFragment : Fragment() {
+abstract class BaseFragment : DaggerFragment(), AnalyticScreen {
     abstract val layoutResource: Int
+    override val screen: Screen? = null
+
+    @Inject
+    lateinit var analyticsManager: AnalyticsManager
 
     override fun onCreateView(
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/common/base/BaseFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/669806ff547ce590f35db7d750b40abf49b781bc/app/src/main/kotlin/co/netguru/baby/monitor/client/common/base/BaseFragment.kt#L19"></span>Parameter should be on a separate line (unless all parameters can fit a single line)</td>
    </tr>
  </tbody>
</table>

**Pull request merged**